beprof/curve.py

```diff
@@ -66,19 +66,74 @@ def y_at_x(self, x):
             return x
         return np.interp(x, self.x, self.y, left=np.nan, right=np.nan)
 
+    def change_domain(self, domain):
+
+        # check if new domain includes in the orginal domain
+        if np.max(domain) > np.max(self.x):
+            # maybe some exception here or stderr log ?
+            # these prints only for testing purpose
+            print('Error1')
+            print('domain.max: ', np.max(domain), '|| self.max: ', np.max(self.x))
+            return self
+        if np.min(domain) < np.min(self.x):
+            print('Error2')
+            print('domain.min: ', np.min(domain), '|| self.min: ', np.min(self.x))
+            return self
+        y = np.interp(domain, self.x, self.y)
+        obj = Curve([[domain[ind], y[ind]] for ind in range(0, len(y))])
+        return obj
+
+    def fixed_step_domain(self, step=0.1, fixp=0):
+        section = (np.min(self.x), np.max(self.x))
+        count_start = (abs(fixp - section[0]) / step)
+        count_stop = (abs(fixp - section[1]) / step)
+
+        # depending on position of fixp with respect to the orginal domain
+        # may be 1 of 3 cases:
+
+        if fixp < section[0]:
+            count_start = int(np.ceil(count_start))
```

Comment:
Why are you using numpy ceil instead of math ceil ?
It seems math is faster:
```

grzanka@gajewski-dell:~$ python3 -m timeit -r 10 -s "import math" "math.ceil(2.5)"
10000000 loops, best of 10: 0.0942 usec per loop
grzanka@gajewski-dell:~$ python3 -m timeit -r 10 -s "import numpy as np" "np.ceil(2.5)"
1000000 loops, best of 10: 0.512 usec per loop

```